crates/core/src/surfnet/svm.rs

```diff
@@ -414,6 +422,96 @@ impl SurfnetSvm {
         }
     }
 
+    pub fn simulate_with_account_snapshots(
+        &self,
+        _remote_ctx: &Option<SurfnetRemoteClient>,
```

Comment:
This is close!

However, you'll see here that we're not using the `remote_ctx` - this is how we get data from the mainnet data source.

So, if the user sends a transaction to be simulated that doesn't have all of the accounts available in the local state, the transaction will fail.

You can see down below, you're doing `svm_clone.inner.get_account` - this means you're getting just the local account.

If you do `let svm_locker_clone = SurfnetSvmLocker::new(svm_clone);`, then you should be able to properly process transactions:

`svm_locker.process_transactions(...)`

This will require you to pass in this remote context, and will pull the accounts from the remote as needed